central/scanner/handler_test.go

```diff
@@ -45,8 +45,9 @@ func (s *handlerTestSuite) TestGenerateScannerHTTPHandler() {
 	s.Assert().Equal(http.StatusOK, resp.StatusCode)
 	body, err := io.ReadAll(resp.Body)
 	s.Require().NoError(err)
-	s.Assert().NotEmpty(body)
+	s.Require().NoError(resp.Body.Close())
```

Comment:
Should we put defer on top such that we close even if the requires fail?

Reply:
Do we have anything to close if err != nil? I think we need to be careful because `Body` might be nil in case of error